File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.*;

class Solution {
    public int solution(int[] scoville, int K) {
        int answer = 0;
        PriorityQueue<Integer> pq=new PriorityQueue<>();
        for(int i = 0;i<scoville.length;i++){
            pq.offer(scoville[i]);
        }
        int sco=0;
        int cnt=0;
        boolean flag=true;
        while(!pq.isEmpty() && pq.peek()<K){
            if(pq.size()<2){
                return -1;
            }
            int first=pq.poll();
            int second=pq.poll();
            pq.offer(first+2*second);
            
            cnt++;
        }
        
        return cnt;
    }
}

File: 프로그래머스/2/42626. 더 맵게/더 맵게.py
import heapq

def solution(scoville, K):
    min_hq=[]
    heapq.heapify(scoville)
    cnt=0
    min_hq=scoville
    flag=True
    while min_hq and min_hq[0]<K:
        sco=heapq.heappop(min_hq)
        if min_hq:
            
            sco2=heapq.heappop(min_hq)
            new_sco=sco+2*sco2
            heapq.heappush(min_hq,new_sco)
            cnt+=1
        else:
            flag=False
            break
        #print(min_hq[0])
    if flag:
        return cnt
    else:
        return -1
    